### PR TITLE
PIMOB-422: fix dependency structures

### PR DESCRIPTION
**Pull request by harry-brown-cko**
## Proposed changes

- Move project dependencies to Cocoapods and Swift Packages
- Setup example projects using Cocoapods and Swift Packages
- Bump `Alamofire` and `PhoneNumberKit` to latest releases
- Add Cocoapods and Swift Packages verification steps to CI

## Types of changes

What types of changes does your code introduce to Frames Ios?
_Put an `x` in the boxes that apply_

* [X] Bugfix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

* [X] Lint and unit tests pass locally with my changes
* [x] I have added tests that prove my fix is effective or that my feature works
* [ ] I have added necessary documentation (if appropriate)

## Further comments

- Due to Swift Package Manager setup on the sample project we will need a follow up PR to set the branch from `bug/PIMOB-422-fix-dependency-structures` to `develop`
- Should address the following issues:
    - https://github.com/checkout/frames-ios/issues/46
    - https://github.com/checkout/frames-ios/issues/38

### CODE REVIEW

File: Frames.podspec [hunk 1]
```diff
@@ -15,8 +15,15 @@ Pod::Spec.new do |s|
 
   s.source_files = 'Source/**/*.swift'
   s.exclude_files = "Classes/Exclude"
-  s.resource_bundles = {
-    'FramesIos' => 'Source/Resources/**/*'
-  }
+  s.resources = 'Source/Resources/**/*'
 
+  s.dependency 'PhoneNumberKit', '~> 3.3.0'
+  s.dependency 'Alamofire', '~> 5.4.0'
```
Comment (deepesh-vasthimal-cko) [inline Frames.podspec:21]: we can get rid of this as we dont use it.

File: Frames.podspec [hunk 2]
```diff
@@ -15,8 +15,15 @@ Pod::Spec.new do |s|
 
   s.source_files = 'Source/**/*.swift'
   s.exclude_files = "Classes/Exclude"
-  s.resource_bundles = {
-    'FramesIos' => 'Source/Resources/**/*'
-  }
+  s.resources = 'Source/Resources/**/*'
 
+  s.dependency 'PhoneNumberKit', '~> 3.3.0'
```
Comment (deepesh-vasthimal-cko) [inline Frames.podspec:20]: 👍

File: Source/CheckoutAPIClient.swift [hunk 3]
```diff
@@ -1,4 +1,5 @@
 import Foundation
+import Alamofire
```
Comment (deepesh-vasthimal-cko) [inline Source/CheckoutAPIClient.swift:2]: Do we really need this ?

File: Frames.podspec (hunk 1 above)
Comment (harry-brown-cko) [reply Frames.podspec:21]: no it is in use

File: Source/CheckoutAPIClient.swift (hunk 3 above)
Comment (harry-brown-cko) [reply Source/CheckoutAPIClient.swift:2]: yes its being used for requests

**Review by daven-gomes-cko — APPROVED** [review]

**Review by deepesh-vasthimal-cko — APPROVED** [review]

**Pull request merged**